src/spec/models/tag_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe Tag, type: :model do
+  subject { create :tag }
+
+  it { is_expected.to validate_presence_of :name }
+  it { is_expected.to validate_uniqueness_of :name }
+  it { is_expected.to have_and_belong_to_many :functions }
+
+  it { is_expected.to allow_value('tag').for :name }
+  it { is_expected.to allow_value('also_a_tag').for :name }
+  it { is_expected.to allow_value('another-valid-tag').for :name }
```

Comment:
I'm not sure we should allow both `-` and `_` I'm leaning towards more restriction, can we use only `-`? I think it's even more URL friendly as the tag will be part of the URL when showing the tag functions